File: ContinuixV1/src/main/java/tn/sopra/continuix/entities/Alerte.java
package tn.sopra.continuix.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Data

public class Alerte implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String description;
    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    private IncidentType type;
    @Column(name = "impact")
    @Enumerated(EnumType.STRING)
    private Impact impact;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @JoinColumn(name = "user_id")
    private Users createdBy;
    private boolean resolved = false;
    private boolean simulation = false;
}

File: ContinuixFront-endV1/Continuix/src/app/models/alerte.ts
import { Impact } from "../impact";
import { IncidentType } from "../incident-type";
import { User } from "../user";

export interface Alerte {
  id?: number;
  description: string;
   type: string     // si tu veux, tu peux mettre IncidentType | string
  impact: string;   // Impact | string
  createdBy?: User;
   resolved?: boolean;

}
